spec/std/big/big_decimal_spec.cr

```diff
@@ -337,36 +337,43 @@ describe BigDecimal do
     bd2 = BigDecimal.new(-123, 5)
     bd3 = BigDecimal.new(123, 0)
     bd4 = BigDecimal.new(-123, 0)
+    bd5 = "-123.000".to_big_d
 
     bd1.to_i.should eq 0
     bd2.to_i.should eq 0
     bd3.to_i.should eq 123
     bd4.to_i.should eq -123
+    bd5.to_i.should eq -123
 
     bd1.to_u.should eq 0
     bd2.to_u.should eq 0
     bd3.to_u.should eq 123
     bd4.to_u.should eq 123
+    bd5.to_u.should eq 123
```

Comment:
@bcardiff btw, why do `Big*#to_u` methods for values < 0 are returning them as absolute instead of raising?

Reply:
That seems unexpected...

Reply:
I think is the effect of time, when `Int#to_u` used to be more like a cast, and `BigDecimal#to_big_u` simple uses an `.abs` as an approximation. Nobody submit any opinions regarding that in the original PR #4876 .

It should raise `OverflowError` for negative inputs.

Reply:
Was thinking the same, although I'd leave it for another PR.

Reply:
For further consideration: All these specs don't really test what you would expect. Equality between different number types already works without explicit conversation. So these specs should also validate the return type.

Reply:
@straight-shoota Could we leave it for another PR? I don't think we should do much more here.

Reply:
Absolutely

Reply:
btw, `BigInt#to_u` has the same issue:

```cr
require "big"

-123.to_big_i.to_u # => 123
```